src/Core/Compiler/TemplateCompiler.php

```diff
@@ -37,13 +39,49 @@ class TemplateCompiler {
 	protected $renderingContext;
 
 	/**
+	 * @var string
+	 */
+	protected $mode = self::MODE_NORMAL;
```

Comment:
constant to property assignment, wasn't there this impossible until "recently" in php?
do you know, since when this is possible?